boot/bootstate20.go

```diff
@@ -482,62 +568,28 @@ func (ks20 *bootState20Kernel) setNext(next snap.PlaceInfo) (rebootRequired bool
 
 	// if we are setting a snap as a try snap, then we need to reboot
 	rebootRequired = false
-	ks20.nextKernelSnap = next
 	if nextStatus == TryStatus {
 		rebootRequired = true
 	}
-	ks20.commitKernelStatus = nextStatus
 
-	// any state changes done so far are consumed in commit()
-
-	return rebootRequired, ks20, nil
-}
-
-// commit for bootState20Kernel is meant only to be used with setNext().
-// For markSuccessful(), use bootState20MarkSuccessful.
-func (ks20 *bootState20Kernel) commit() error {
-	// The ordering of this is very important for boot safety/reliability!!!
-
-	// If we are about to try an update, and need to add the try-kernel symlink,
-	// we need to do things in this order:
-	// 1. Add the kernel snap to the modeenv
-	// 2. Create try-kernel symlink
-	// 3. Update kernel_status to "try"
-	//
-	// This is because if we get rebooted in before 3, kernel_status is still
-	// unset and boot scripts proceeds to boot with the old kernel, effectively
-	// ignoring the try-kernel symlink.
-	// If we did it in the opposite order however, we would set kernel_status to
-	// "try" and then get rebooted before we could create the try-kernel
-	// symlink, so the bootloader would try to boot from the non-existent
-	// try-kernel symlink and become broken.
-	//
-	// Adding the kernel snap to the modeenv's list of trusted kernel snaps can
-	// effectively happen any time before we update the kernel_status to "try"
-	// for the same reasoning as for creating the try-kernel symlink. Putting it
-	// first is currently a purely aesthetic choice.
-
-	// add the kernel to the modeenv if it is not the current kernel (if it is
-	// the current kernel then it must already be in the modeenv)
 	currentKernel := ks20.bks.kernel()
-	if ks20.nextKernelSnap.Filename() != currentKernel.Filename() {
-		// add the kernel to the modeenv
-		ks20.kModeenv.modeenv.CurrentKernels = append(
-			ks20.kModeenv.modeenv.CurrentKernels,
-			ks20.nextKernelSnap.Filename(),
+	if next.Filename() != currentKernel.Filename() {
+		// add the kernel to the write modeenv
+		u20.writeModeenv.CurrentKernels = append(
```

Comment:
this might need a "On commit..." introduction, same for all the .writeModeenv changes/change groupings